src/test/ui/pattern/bindings-after-at/bind-by-move-neither-can-live-while-the-other-survives-1.rs

```diff
@@ -0,0 +1,12 @@
+#![feature(bindings_after_at)]
+//~^ WARN the feature `bindings_after_at` is incomplete and may cause the compiler to crash
```

Comment:
The filename here makes me think that your intent with this test is to check something about lifetimes and/or dynamic extents. But the current error message is enforcing a much coarser property. Is that your intent? Maybe the test should have a comment explaining what it is trying to enforce?

Reply:
This test is taken directly from https://github.com/rust-lang/rust/pull/16053/files#diff-01fbe0eb37ab35aa74df1ef430b9b455 -- I'll extend it a bit and leave a comment.